src/init_interp/initInterpMultilevelInterp.F90

```diff
@@ -346,11 +369,17 @@ subroutine interp_multilevel(this, data_dest, data_source, index_dest)
              end do
              call pack_wrapper(data_source_in_class, data_source, source_levels_in_class)
              call pack_wrapper(coordinates_source_in_class, my_coordinates_source, source_levels_in_class)
+             call pack_wrapper(dzsoi_source_in_class, my_dzsoi_source, source_levels_in_class)
+             dzsoi_dest_sum = dzsoi_dest_sum + this%dzsoi_dest(lev_dest, index_dest)
```

Comment:
`dzsoi_dest_sum` is a new variable in the `scale_by_thickness=.true.` calculation.